### PR TITLE
Switched OS media controls

### PR DESCRIPTION
**Pull request by ferferga**
Jellyfin was the only app I knew that displayed the artist's name as the title of the item in the OS media controls of all the devices I have. I don't know if that was intended that way, but how I left it makes much more sense to me:

* **Before**

Screenshot_20200126_000327_com android chrome
Screenshot_20200126_000332_com android keyguard
media_controls2

* **After**

Screenshot_20200126_000150_com huawei android launcher
Screenshot_20200126_000155_com android keyguard
media_controls

### CODE REVIEW

File: src/components/playback/mediasession.js [hunk 1]
```diff
@@ -160,8 +160,8 @@ define(['playbackManager', 'nowPlayingHelper', 'events', 'connectionManager'], f
 
         if (navigator.mediaSession) {
             navigator.mediaSession.metadata = new MediaMetadata({
-                title: title,
-                artist: artist,
+                title: artist,
+                artist: title,
```
Comment (dkanada) [inline src/components/playback/mediasession.js]: I would rather find the code that inverts the logic and remove that.

**Comment by ferferga** [comment]
Logic fixed. I also noticed that updates were received, processed and discarded with a return if there were no changes. I moved that check to the top of the logic, so it does check it first, which seems more logical to me.

File: src/components/playback/mediasession.js [hunk 2]
```diff
@@ -118,19 +127,9 @@ define(['playbackManager', 'nowPlayingHelper', 'events', 'connectionManager'], f
...
 
         var playState = state.PlayState || {};
-
         var parts = nowPlayingHelper.getNowPlayingNames(item);
-
-        var artist = parts.length === 1 ? '' : parts[0].text;
-        var title = parts[parts.length - 1].text;
-
-        // Switch these two around for video
-        if (isVideo && parts.length > 1) {
```
Comment (JustAMan) [inline src/components/playback/mediasession.js]: do you have any clue on why was this ever needed?..

Comment (ferferga) [reply src/components/playback/mediasession.js]: No idea. It was right, however, given how the variables were first declared, don't know why they didn't do the same for music 😆.

**Comment by anthonylavado** [comment]
Maybe this is what was happening to @Artiume in this issue/PR: https://github.com/jellyfin/jellyfin-android/pull/234

**Comment by ferferga** [comment]
99% sure this is the solution. For all devices that support media controls, not only for Bluetooth audio though.

File: src/components/playback/mediasession.js [hunk 3]
```diff
@@ -97,6 +97,15 @@ define(['playbackManager', 'nowPlayingHelper', 'events', 'connectionManager'], f
     }
 
     function updatePlayerState(player, state, eventName) {
+        var now = new Date().getTime();
+        lastUpdateTime = now;
+
+        // Don't go crazy reporting position changes
+        if (eventName == 'timeupdate' && (now - lastUpdateTime) < 5000) {
```
Comment (marius-luca-87) [inline src/components/playback/mediasession.js]: lastUpdateTime = now;
now - lastUpdateTime = 0

**Comment by Artiume** [comment]
Thanks for finding the fix :)

File: src/components/playback/mediasession.js (hunk 3 above)
Comment (ferferga) [reply src/components/playback/mediasession.js]: Wow, completely missed that. It's crazy to see what we got from Emby codebase!

Thanks for pointing it out!

**Comment by ferferga** [comment]
@marius-luca-87 For some reason, clicking on the re-request button doesn't do anything, so I can't notify you. Can you review this again, please?

**Pull request merged**